mkosi

```diff
@@ -1027,8 +1027,14 @@ def reset_machine_id(args, workspace, run_build_script):
         os.unlink(machine_id)
         open(machine_id, "w+b").close()
         dbus_machine_id = os.path.join(workspace, 'root', 'var/lib/dbus/machine-id')
-        if os.path.exists(dbus_machine_id):
+        has_dbus_machine_id = False
+        try:
             os.unlink(dbus_machine_id)
+            has_dbus_machine_id = True
+        except FileNotFoundError:
+            pass
+
+        if has_dbus_machine_id:
```

Comment:
Actually, what I want is to create the symlink only if the file already existed. I do not want to create it if we didn't remove it. That would imply creating the `/var/lib/dbus` dir if it doesn't exist, which I don't think we should be doing. That leaves us with either this PR or:

```python
try:
  os.unlink(dbus_machine_id)
  os.symlink('../../../etc/machine-id', dbus_machine_id)
except FileNotFoundError:
  pass
```